proposals/7-payload-processing.md

```diff
@@ -165,18 +165,101 @@ Requirements:
->
-> This section is fairly freeform, because (again) these proposals will
-> eventually find there way into any number of different final proposal formats
-> in other projects. However, the general guidance is to break things down into
-> highly focused sections as much as possible to help make things easier to
-> read and review. Long, unbroken walls of code and YAML in this document are
-> not advisable as that may increase the time it takes to review.
+
+
+### Operational Requirements and Assumptions
+
+* Gateway MUST provide Processors with plaintext traffic. Any decryption and (re-)encryption MUST happen in Gateway implementation.
+* Gateway MUST support Processors running as workloads on the cluster, as well as remote endpoints. (We should probably focus on cluster-local Services for the first phase)
+* Topology-Aware Routing (or similar constraints) are highly desirable when configuring Gateway consumption of Processors (both due to latency
+concerns, and also since cross-node/cross-AZ traffic may have cloud-networking costs associated with it).
+* Processor MAY support scale based on operational metrics from Gateway.
+* Processor MAY present indications of its capacity (or lack thereof) to Gateway. Gateway MAY support reducing the traffic 
+load on the Processor if such indications are presented. In some scenarios, the user MAY prefer degrading payload processing
+over significant impact on data plane performance.
+
+## Resource Model
+
+
+### Payload Processor Custom Resource
+
+
+`HTTPRoute` **rules** are the most specific level where we wish to prescribe Payload Processors.
+The amount of details and settings we need to support at the processor level is greater than what we can comfortable inline within a sub-object
+of `HTTPRoute` (See also how long `spec.rules.filters` can become on its own, today).
+
+The approach below creates a new custom resource kind for Payload Processing Pipelines.
+A minimal `HTTPRoute` that references a processor pipeline is shown below.
+
+```yaml
+apiVersion: gateway.networking.k8s.io/v1
+kind: HTTPRoute
+metadata:
+  name: example-httproute
+spec:
+  parentRefs:
+  - name: example-gateway
+  hostnames:
+  - "www.example.com"
+  rules:
+  - matches:
+    - path:
+        type: PathPrefix
+        value: /query
+    backendRefs:
+    - name: example-svc
+      port: 8080
+    processing:
+    - name: prompt-injection-guardrails
+      pipelineRef:
+        name: prompt-injection-protection
+---
+apiVersion: gateway.networking.k8s.io/v1alpha1
+kind: PayloadProcessingPipeline
+metadata:
+  name: prompt-injection-protection
+spec:
+  # Sequence of processors
+  rules:
+  - name: malicious-prompt-prevention
+    # Indicates if this processor rule can mutate passed data
+    mutating: true
+    # Determines which parts of the request/response pair is of interest to the processor
+    phases: ["request-headers", "request-body", "response-headers", "response-body"]
```

Comment:
How is this different from HTTPRoute's filter https://gateway-api.sigs.k8s.io/reference/spec/#httproutefilter